aten/src/ATen/native/Normalization.cpp

```diff
@@ -188,18 +188,19 @@ std::tuple<Tensor,Tensor> batch_norm_cpu_update_stats_template(
     auto _var_sum_a = _var_sum.accessor<param_t, 1>();
 
     batch_norm_cpu_collect_stats_stub(kCPU, _mean, _var_sum, input);
-
+    auto momentum_ = static_cast<param_t>(momentum);
```

Comment:
Why is the static cast necessary? (what was the composite compliance problem?)

Reply:
This won't work as the computation in `double` results in values out of `float`'s range.

```c++
int main() {
    double d = -5.368e+38;
    float f = static_cast<float>(d);
}
```

Following code compiled as `clang++ test_ubsan.cc -fsanitize=undefined`
Fails with
```
test_ubsan.cc:3:15: runtime error: -5.368e+38 is outside the range of representable values of type 'float'
    #0 0x4242f1  (/home/kshiteej/Pytorch/pytorch_functorch/test/a.out+0x4242f1)
    #1 0x7fef65587d8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f)
    #2 0x7fef65587e3f  (/lib/x86_64-linux-gnu/libc.so.6+0x29e3f)
```